tests/test_recipient_csv.py

```diff
@@ -162,6 +162,36 @@ def test_get_rows(file_contents, template_type, expected):
             assert rows[index].get(key).data == value
 
 
+def test_get_rows_does_no_error_checking_of_rows_or_cells(mocker):
+    recipients = RecipientCSV(
+        """
+            email address, name
+            a@b.com,
+            a@b.com, My Name
+            a@b.com,
+
+
+        """,
+        template_type='email',
+        placeholders=['name'],
+        max_errors_shown=3
+    )
+
+    has_error_mock = mocker.patch.object(Row, 'has_error')
+    has_bad_recipient_mock = mocker.patch.object(Row, 'has_bad_recipient')
+    has_missing_data_mock = mocker.patch.object(Row, 'has_missing_data')
+    cell_recipient_error_mock = mocker.patch.object(Cell, 'recipient_error')
+
+    rows = recipients.get_rows()
+    for i in range(3):
+        next(rows)
```

Comment:
This would be a more realistic test of what the code in the API is doing:
```suggestion
        assert next(rows).recipient == 'a@b.com'
```